### PR TITLE
Correct BlackSkull spells limitations

### LINKED ISSUE #2329: No BlackSkull spells limitations
### Before creating an issue, please ensure:
- [+] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [+] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Cast an area spell

### Expected behaviour
It should return cancel message saying you may not use area spells on BlackSkull

### Actual behaviour
There is no limit for spells on blackskull

### PR DESCRIPTION
**Pull request by nekiro**
This pull request should fix #2329 

According to http://tibia.wikia.com/wiki/Skull_System

> When a player has a black skull they will no longer be able to attack an unmarked player - that is any player without a skull. In addition to this, a black-skulled player may not use area attack spells and runes (i.e. Hell's Core, Explosion, etc.), as well as Burst Arrows to prevent black skulls from killing unmarked players indirectly. They also cannot summon or convince creatures, or use Undead Legion or strike spells (i.e. Energy Strike) which are not aimed at a target.

### CODE REVIEW

File: data/spells/scripts/support/summon_creature.lua [hunk 1]
```diff
@@ -1,4 +1,9 @@
 function onCastSpell(creature, variant)
+	if player:getSkull() == SKULL_BLACK then
```
Comment (wgrriffel) [inline data/spells/scripts/support/summon_creature.lua]: player nil value here

Comment (thexamx) [reply data/spells/scripts/support/summon_creature.lua]: thanks!

File: data/spells/scripts/support/summon_creature.lua [hunk 2]
```diff
@@ -7,7 +12,7 @@ function onCastSpell(creature, variant)
 		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
 		return false
 	end
-
+
```
Comment (ranisalt) [inline data/spells/scripts/support/summon_creature.lua]: Trailing whitespace

File: data/weapons/scripts/burst_arrow.lua [hunk 3]
```diff
@@ -13,5 +13,9 @@ combat:setFormula(COMBAT_FORMULA_SKILL, 0, 0, 1, 0)
 combat:setArea(area)
 
 function onUseWeapon(player, variant)
+	if player:getSkull() == SKULL_BLACK then
+		return false
+	end
+
```
Comment (ranisalt) [inline data/weapons/scripts/burst_arrow.lua]: Trailing whitespace

File: src/monster.cpp [hunk 4]
```diff
@@ -1915,6 +1915,10 @@ bool Monster::challengeCreature(Creature* creature)
 bool Monster::convinceCreature(Creature* creature)
 {
 	Player* player = creature->getPlayer();
+	if (player->getSkull() == SKULL_BLACK) {
+		return false;
+	}
+
```
Comment (ranisalt) [inline src/monster.cpp]: Same

File: src/spells.cpp [hunk 5]
```diff
@@ -560,6 +560,11 @@ bool Spell::playerSpellCheck(Player* player) const
 		return false;
 	}
 
+	if (aggressive && (range < 1 || (range > 0 && !player->getAttackedCreature())) && player->getSkull() == SKULL_BLACK) {
+		player->sendCancelMessage(RETURNVALUE_NOTPOSSIBLE);
+		return false;
+	}
+
```
Comment (ranisalt) [inline src/spells.cpp]: Same

File: data/spells/scripts/support/summon_creature.lua (hunk 2 above)
Comment (ranisalt) [reply data/spells/scripts/support/summon_creature.lua]: Still missing

**Comment by WibbenZ** [comment]
@nekiro Can you resolve the conflicts?

**Pull request merged**